### PR TITLE
Add instances test

### PR DESCRIPTION
**Pull request by sirococoa**
## Summary

- Add test for methods of nifcloud_instances.go.
- Fix to use isAPIError for handling error of computing API
  - 529c42776d3410c588f06cdac794e54e74b3f49d

## Review

- [ ] CI is successful.

### CODE REVIEW

File: pkg/cloudprovider/providers/nifcloud/nifcloud_instances.go [hunk 1]
```diff
@@ -139,7 +138,7 @@ func (c *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID str
 func (c *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
 	_, err := c.getInstance(ctx, node)
 	if err != nil {
-		if errors.Is(err, cloudprovider.InstanceNotFound) {
+		if isAPIError(err, errorCodeInstanceNotFound) {
```
Comment (aokumasan) [inline pkg/cloudprovider/providers/nifcloud/nifcloud_instances.go:141]: The NotFound API error is handled in client code and it converts to `cloudprovider.InstanceNotFound` such as:
- https://github.com/nifcloud/nifcloud-cloud-controller-manager/blob/7e9f9f1a615adf2ce241fe1d82ec5547ebc2bf02/pkg/cloudprovider/providers/nifcloud/nifcloud_client.go#L220-L222
- https://github.com/nifcloud/nifcloud-cloud-controller-manager/blob/7e9f9f1a615adf2ce241fe1d82ec5547ebc2bf02/pkg/cloudprovider/providers/nifcloud/nifcloud_client.go#L187-L189

So I think this change breaks the logic of error handling.

Comment (sirococoa) [reply pkg/cloudprovider/providers/nifcloud/nifcloud_instances.go]: I forgot that `nifcloudAPIClient.DescribeInstancesByInstanceID` and `nifcloudAPIClient.DescribeInstancesByInstanceUniqueID` return `cloudprovider.InstanceNotFound` error when the instance is not found.

So, I fix all mock errors show an instance is not found to `cloudprovider.InstanceNotFound`.

ff32d88295ba4659012bffce9aec55030e5853b7

**Review by aokumasan — APPROVED** [review]

File: pkg/cloudprovider/providers/nifcloud/nifcloud_instances.go (hunk 1 above)
Comment (aokumasan) [reply pkg/cloudprovider/providers/nifcloud/nifcloud_instances.go]: Thanks!

**Comment by aokumasan** [comment]
LGTM 🐱

**Pull request merged**